sitegen/src/main/java/io/helidon/sitegen/asciidoctor/Block.java

```diff
@@ -0,0 +1,351 @@
+import java.util.List;
+import java.util.Set;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.function.Consumer;
+import java.util.function.Predicate;
+import java.util.function.Supplier;
+import java.util.regex.Matcher;
+
+/**
+ * Describes content that is a block (e.g., {@code [source]}) that might contain
+ * includes.
+ * <p>
+ * <h1>Overview</h1>
+ * For our purposes a block contains
+ * <ul>
+ * <li>its declaration (e.g., {@code [source]} with, potentially, additional
+ * attributes),
+ * <li>any {@code include::} directives that contribute to the body of the
+ * block, and
+ * <li>the body itself.
+ * </ul>
+ * <h1>Text Formatting</h1>
+ * At different times we represent a block as text in three ways.
+ * <h2>Brief Bracketed Form (internal)</h2>
+ * In this form we bracket each AsciiDoc {@code include::} directive in the body
+ * with a preceding {@code // _include-start::} and a following
+ * {@code // _include-end::} comment. Our preprocessing submits this form to
+ * AsciiDoc in a way that creates the full bracketed form.
+ * <h2>Full Bracketed Form (internal)</h2>
+ * In this format the bracketing comments surround the included text itself
+ * rather than the {@code include::} AsciiDoc directive.
+ * <h2>Numbered Form</h2>
+ * In this format:
+ * <ul>
+ * <li>The preamble contains an AsciiDoc comment for each AsciiDoc
+ * {@code include::} that originally appeared in the body. Each such comment
+ * indicates the origin of the include and the line numbers within the block's
+ * body where the corresponding included content resides.
+ * <li>The body contains the included content in the correct positions.
+ * </ul>
+ * For example, given an original AsciiDoc block like this:
+ * <pre>
+ * {@code
+ * [source]
+ * .Title for the source
+ * ----
+ * // This is un-included source
+ * include::Include1.adoc
+ * More un-included source
+ * include::Include2.adoc
+ * Further content
+ * ----
+ * }
+ * </pre> here is the corresponding numbered form:
+ * <pre>
+ * {@code
+ * [source]
+ * .Title for the source
+ * // _include::1-4:Include1.adoc
+ * // _include::6-9:Include2.adoc
+ * ----
+ * // This is un-included source
+ * src inc 1.1
+ * src inc 1.2
+ * src inc 1.3
+ * src inc 1.4
+ * More un-included source
+ * src inc 2.1
+ * src inc 2.2
+ * src inc 2.3
+ * src inc 2.4
+ * Further content
+ * ----
+ * }
+ * </pre>
+ * <h2>Creating {@code Block} Instances by Parsing AsciiDoc</h2>
+ * The {@link #consumeBlock} method reads content and builds an instance
+ * representing the corresponding block, advancing the current line indicator
+ * for the content as it does so.
+ * <p>
+ * Note that this method handles the original AsciiDoc form (with
+ * {@code include::} directives in the body), pure numbered format, and a hybrid
+ * in which the user has added AsciiDoc {@code include::} directives in the body
+ * of the numbered form.
+ */
+public class Block {
+
+    private static final Set<String> BLOCK_DELIMITERS
```

Comment:
This could be a Stream and created with Stream.of("", "")

Reply:
Actually, now that you've caused me to look at that code again, I think I'll change the uses of those two constants to use `XXX.contains` instead of using streams with `anyMatch`.

Anyway, I think declaring and instantiating the stream using `Stream.of` would mean we could only use it once, but we need to be able to check for one of the contained values multiple times, so I think we'd get stream-already-operated-upon errors.

Reply:
Right, I missed the point that streams can be used just once.